wondrous-bot-admin/src/components/AddFormEntity/components/ComponentOptionsModal/ComponentOptionsModal.tsx

```diff
@@ -0,0 +1,97 @@
+import { Button, Grid, Stack, Typography, useTheme } from "@mui/material";
+import { COMPONENT_CATEGORIES } from "components/AddFormEntity/constants";
+import SafeImage from "components/SafeImage";
+import Modal from "components/Shared/Modal";
+import { CategoryTitle, ColumnContainer, OptionButton, OptionButtonLabel, OptionsColumnContainer } from "./styles";
+
+const chooseIcon = (icon, category) => {
+  if (!icon) {
+    // in case icon is not provided, particularly for custom integrations, return a default icon
+    return <img src={"/wonder.svg"} alt={icon} style={{ width: "25px", height: "25px", borderRadius: "6px" }} />;
+  }
+  if (category === COMPONENT_CATEGORIES.CUSTOM) {
+    return (
+      <SafeImage
+        src={icon}
+        alt="icon"
+        style={{ width: "25px", height: "25px", borderRadius: "6px", objectFit: "cover" }}
+      />
+    );
+  }
+  return <img src={icon} alt={icon} />;
+};
+
+const OptionsColumn = ({ colOptions, options, onClick, onClose, showBorder }) => {
+  return (
+    <OptionsColumnContainer $showBorder={showBorder}>
+      {colOptions.map((category) => (
+        <Stack spacing={1} key={category}>
+          <CategoryTitle>{category}</CategoryTitle>
+          <Stack alignItems="start">
+            {options[category].map((option) => (
+              <OptionButton
+                key={option.value}
+                disableRipple
+                onClick={() => {
+                  onClick(option.value);
+                  onClose();
+                }}
+                startIcon={chooseIcon(option.icon, category)}
+              >
+                <OptionButtonLabel>{option.label}</OptionButtonLabel>
+              </OptionButton>
+            ))}
+          </Stack>
+        </Stack>
+      ))}
+    </OptionsColumnContainer>
+  );
+};
+
+const ComponentOptionsModal = ({ open, onClose, onClick, options }) => {
+  const groupedOptionsByCategory = options.reduce((acc, option) => {
+    if (!acc[option.category]) {
+      acc[option.category] = [];
+    }
+    acc[option.category].push(option);
+    return acc;
+  }, {});
+
+  const leftColOptions = [COMPONENT_CATEGORIES.ACTION, COMPONENT_CATEGORIES.TWITTER];
+  const midColOptions = [COMPONENT_CATEGORIES.DISCORD, COMPONENT_CATEGORIES.YOUTUBE, COMPONENT_CATEGORIES.CRYPTO];
+  const rightColOptions = [COMPONENT_CATEGORIES.CUSTOM];
+
+  const hasCustomComponent = options.some((option) => option.category === COMPONENT_CATEGORIES.CUSTOM);
```

Comment:
let's include this one and the reduce from above into a memo, these two can probably be combined